pkg/ext-proc/backend/llmserverpool_controller.go

```diff
@@ -0,0 +1,87 @@
+package backend
+
+import (
+	"context"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"inference.networking.x-k8s.io/llm-instance-gateway/api/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	controllerNamePrefix = "instance-gateway-"
+)
+
+// LLMServerPoolController is the controller implementation for Instance Gateway resources
+type LLMServerPoolController struct {
+	client.Client
+	Scheme         *runtime.Scheme
+	Record         record.EventRecorder
+	ServerPoolName string
+	Namespace      string
+	Datastore      *K8sDatastore
+	Port           int
+	Zone           string
+}
+
+func (c *LLMServerPoolController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	if req.NamespacedName.Name != c.ServerPoolName && req.NamespacedName.Namespace != c.Namespace {
+		return ctrl.Result{}, nil
+	}
+	klog.V(1).Info("reconciling LLMServerPool", req.NamespacedName)
+
+	serverPool := &v1alpha1.LLMServerPool{}
+	if err := c.Get(ctx, req.NamespacedName, serverPool); err != nil {
+		klog.Error(err, "unable to get LLMServerPool")
+		return ctrl.Result{}, err
+	}
+
+	needNewService := true
+	if needNewService {
+		labels, err := metav1.LabelSelectorAsMap(&c.Datastore.LLMServerPool.Spec.ModelServerSelector)
+		if err != nil {
+			klog.Error(err, "error converting label selector to map.")
+		}
+		service := &v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: controllerNamePrefix + c.ServerPoolName,
+				//figure out ownership labels
+			},
+			Spec: v1.ServiceSpec{
+				Ports: []v1.ServicePort{
+					{
+						Port: int32(c.Port),
+					},
+				},
+				Selector: labels,
+			},
+		}
+		if err := c.Create(ctx, service); err != nil {
+			klog.Error(err, "unable to create service")
+			return ctrl.Result{RequeueAfter: time.Second * 5}, err
```

Comment:
do we need to behave differently if the error indicates that the service exists already?